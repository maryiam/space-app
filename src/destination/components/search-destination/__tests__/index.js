import renderer from 'react-test-renderer';
import React from 'react';
import SearchDestination from '../index';

describe('search-destination/', () => {
  describe('#render', () => {
    it('renders correctly with provided props', () => {
      const tree = renderer.create(<SearchDestination handleChange={jest.fn} />);
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });

  describe('callback invocation', () => {
    it('invokes the function passed as prop when the input value changes', () => {
      const mockFn = jest.fn();
      const tree = renderer.create(<SearchDestination handleChange={mockFn} />);
      const rootInstance = tree.root;
      const input = rootInstance.findByType('input');
      const value = { target: { value: 'abc' } };

      input.props.onChange(value);

      expect(mockFn).toBeCalledWith(value);
    });
  });
});
