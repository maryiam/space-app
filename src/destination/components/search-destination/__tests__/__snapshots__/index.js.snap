// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search-destination/ #render renders correctly with provided props 1`] = `
<input
  className="sc-bdVaJa bYooEG"
  onChange={[Function]}
  placeholder="Search for a planet"
  type="text"
/>
`;
